From a6558402c6cf35dfc574b2dd52f5b3382611d7b9 Mon Sep 17 00:00:00 2001
From: Marcus Brandenburger <bur@zurich.ibm.com>
Date: Wed, 8 Jul 2020 10:23:33 +0200
Subject: [PATCH 1/1] Add FPC validator to library

This adds the FPC validator to the default library of Fabric validation
pipeline and thereby enables to use FPC validator when compiled with the
peer.

Signed-off-by: Marcus Brandenburger <bur@zurich.ibm.com>
---
 core/handlers/library/fpc_lib_handler.go | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)
 create mode 100644 core/handlers/library/fpc_lib_handler.go

diff --git a/core/handlers/library/fpc_lib_handler.go b/core/handlers/library/fpc_lib_handler.go
new file mode 100644
index 000000000..92a355522
--- /dev/null
+++ b/core/handlers/library/fpc_lib_handler.go
@@ -0,0 +1,23 @@
+//+build FPC
+
+package library
+
+import (
+	validation "github.com/hyperledger/fabric/core/handlers/validation/api"
+	decoration "github.com/hyperledger/fabric/core/handlers/decoration"
+	"github.com/hyperledger/fabric/core/handlers/validation/fpc"
+	"github.com/hyperledger/fabric/core/handlers/validation/fpc/ercc"
+)
+
+// Adds FPCValidation to HandlerLibrary if FPC build tag is enabled
+func (r *HandlerLibrary) FPCValidation() validation.PluginFactory {
+	return &fpc.FPCValidationFactory{}
+}
+
+func (r *HandlerLibrary) ERCCValidation() validation.PluginFactory {
+	return &ercc.ERCCValidationFactory{}
+}
+
+func (r *HandlerLibrary) ERCCDecorator() decoration.Decorator {
+	return ercc.NewDecorator()
+}
-- 
2.27.0

